src/LanguageServer/Impl/LanguageServer.cs

```diff
@@ -377,19 +380,29 @@ public Task ExtensionCommand(JToken token, CancellationToken cancellationToken)
                 // Were not watching OR were watching but paths have changed. Recreate the watcher.
                 _pathsWatcher?.Dispose();
                 var interpreter = _services.GetService<IPythonInterpreter>();
+                var logger = _services.GetService<ILogger>();
                 _pathsWatcher = new PathsWatcher(
                     _initParams.initializationOptions.searchPaths,
-                    () => interpreter.ModuleResolution.ReloadAsync(cancellationToken).DoNotWait(),
-                    _services.GetService<ILogger>()
-                 );
+                    () => {
+                        logger.Log(TraceEventType.Information, Resources.ReloadingModules);
+                        interpreter.ModuleResolution.ReloadAsync(cancellationToken).ContinueWith(t => {
```

Comment:
Should we clean the `ModuleResolutionBase._modules` when we do `ReloadAsync`?

Reply:
`ReloadAsync` technically should be doing it, let me check